include/psa/crypto_struct.h

```diff
@@ -223,6 +223,29 @@ static inline struct psa_key_derivation_s psa_key_derivation_operation_init(
     return v;
 }
 
+struct psa_key_generation_method_s {
+    /* Future versions may add other fields in this structure. */
+    uint32_t flags;
+    uint8_t data[];
```

Comment:
Why have a variable-length part at all? Because it seems potentially useful in general. Also, for the case of a custom RSA public exponent, this is variable-length in the most general case. Many RSA implementations only allow smallish public exponent (e.g. Mbed TLS limits it to `INT_MAX`), but I think it would be gratuitously limiting to only allow small values here, since some RSA implementations do allow larger values.